app/src/pages/api/webhooks/btc-pay-server/index.ts

```diff
@@ -1,29 +1,86 @@
+/* eslint-disable @typescript-eslint/naming-convention */
 // Next.js API route support: https://nextjs.org/docs/api-routes/introduction
 
 import { NextApiRequest, NextApiResponse } from "next";
-// import databaseConnection from "src/providers/database";
+import crypto from "crypto";
+import databaseConnection from "src/providers/database";
+import { BusinessFundingCampaignTransactionsModelArgs } from "@bancosatoshi/database/business/model";
 
 import { BtcPayServerWebhookRequestBody } from "./btc-pay-server.types";
+import getInvoicePaymentMethod from "providers/btcpay/getInvoicePaymentMethod";
 
-const headerSignatureMatchesWebhookKey = () => true;
+const bitcoin = require("bitcoin-units");
+
+const headerSignatureMatchesWebhookKey = (sig: string, body: BtcPayServerWebhookRequestBody) => {
+  // @TODO we'll probably need to create a map of storeId => webhook_btcpay_sig because each store has its own webhook key
+  // sha256=HMAC256(UTF8(webhook's secret), body)
+  const hmac = crypto.createHmac("sha256", process.env.WEBHOOKS_BTCPAY_SIG!);
```

Comment:
NOTE: This will fail for other BTCPay stores. 

We need to implement a Secrets manager here.